internal/api/server/http_socket.go

```diff
@@ -14,9 +15,19 @@ import (
 
 var apiServer *http.Server
 
-const API_SERVER_STOP_LOCK = "internal.api.server.API_STOP_LOCK"
+const API_SERVER_STOP_LOCK = "internal.api.server.API_SERVER_STOP_LOCK"
+const PORT_USAGE_CHECK_LOCK = "internal.api.server.PORT_USAGE_CHECK_LOCK"
 
 func runAPIServer(routes *[]core.Route) (err error) {
+	var port = config.WEB_PORT.Get()
+
+	common.Lock(PORT_USAGE_CHECK_LOCK)
+	err = checkPortInUse(port)
+	if err != nil {
+		return err
+	}
+	common.Unlock(PORT_USAGE_CHECK_LOCK)
```

Comment:
There are a couple issues here:
1) You will never unlock on an error.
2) Why not lock inside the function itself, and then you can defer an unlock?
3) Why use a different key than the rest of the file.

!!! (This part is very important.)
I think you may have misunderstood that I asked for before.
I said to error if the server is being used and someone calls runAPIServer(),
and I said to check this you would need to use a lock and manually unlock (not defer) around the check so that you allow people to finish any ongoing calls to stopAPIServer() before you get mad at them.
It is critically important that you don't just listen to what I say but REALLY understand it.
This is not a homework assignment where it is okay for you to miss a few points here and there,
this is a production system that real people rely on.
So it's vital that you can fully stand behind all your code and you can't do that if you don't understand the code and the reasons you are doing it.
The code you have here doesn't make sense on its own,
so my only real conclusion is that you misunderstood what I said and tried to just implement what you thought I said without understanding it.
Ask questions when you need them.

Reply:
Something else to note here, is that if the port is not available the function already returns an error on line 47.
So the port check on line 25 seems unnecessary.